torch/_subclasses/fake_tensor.py

```diff
@@ -444,8 +453,25 @@ def __torch_dispatch__(self, func, types, args=(), kwargs=None):
             else:
                 return args[0].fake_device
 
+        with enable_torch_dispatch_mode(self), torch.overrides.enable_reentrant_dispatch():
```

Comment:
This is another occurrence of https://github.com/pytorch/pytorch/pull/81598#discussion_r924002043

Reply:
Could we get docs for `enable_reentrant_dispatch`? I don't really know what it does tbh.

Reply:
^- @albanD